chore!: rename get senders

Thunkar opened the pull request:
Closes https://linear.app/aztec-labs/issue/F-72/renamegetsenders-to-getaddressbook

Build on top of https://github.com/AztecProtocol/aztec-packages/pull/17835, the diff should be pretty small

benesjan commented on yarn-project/aztec.js/src/wallet/wallet.ts:
```diff
@@ -161,7 +161,7 @@ export type Wallet = {
   getTxReceipt(txHash: TxHash): Promise<TxReceipt>;
   registerSender(address: AztecAddress, alias?: string): Promise<AztecAddress>;
-  getSenders(): Promise<Aliased<AztecAddress>[]>;
+  getAddressBook(): Promise<Aliased<AztecAddress>[]>;
```
I think we should add docs here regarding what this returns and the difference between this and a "sender"

Thunkar replied:
The full interface will be documented soon once the remaining tech debt is removed (I promise)

benesjan commented on yarn-project/aztec.js/src/wallet/base_wallet.ts:
```diff
@@ -87,7 +87,7 @@ export abstract class BaseWallet implements Wallet {
   abstract getAccounts(): Promise<Aliased<AztecAddress>[]>;
 
-  async getSenders(): Promise<Aliased<AztecAddress>[]> {
+  async getAddressBook(): Promise<Aliased<AztecAddress>[]> {
```
I think to a reader it might look weird that we just return here a sender. Would document here that it's a conscious choice and not just a naming issue and that a wallet can decide to return something else.

Thunkar replied:
Explanation added!

benesjan reviewed (approved):
LGTM

Thunkar committed:
chore!: rename get senders

Closes https://linear.app/aztec-labs/issue/F-72/renamegetsenders-to-getaddressbook

Build on top of https://github.com/AztecProtocol/aztec-packages/pull/17835, the diff should be pretty small

Co-authored-by: thunkar <gregojquiros@gmail.com>
